File: frontend/src/services/api/v2/companion.ts
/** Companion ask service — real-model answers for the floating ball
 *  (UX 审计 2026-09-13 P0-A1：此前 ask 路径是罐头演示回复）。 */
import apiClient from '@/services/api/client';
import type { ApiResponse } from '@/types/auth';

export interface CompanionAskRequest {
  context: string;
  question: string;
}

export async function askCompanion(
  data: CompanionAskRequest,
): Promise<ApiResponse<{ answer: string }>> {
  const response = await apiClient.post<ApiResponse<{ answer: string }>>(
    '/companion/ask',
    data,
  );
  return response.data;
}

File: backend/app/api/v2/companion.py
"""Companion ask endpoint (real-model answers for the floating ball)."""

from fastapi import APIRouter, Depends, Request
from pydantic import BaseModel, Field, field_validator

from app.api.deps import get_current_user
from app.models.common import ApiResponse
from app.services.companion import CompanionService

router = APIRouter(prefix="/companion", tags=["Companion"])


class CompanionAskRequest(BaseModel):
    context: str = Field(default="全局", max_length=120)
    question: str = Field(min_length=1, max_length=2000)

    @field_validator("question")
    @classmethod
    def question_not_blank(cls, value: str) -> str:
        if not value.strip():
            raise ValueError("question must not be blank")
        return value


@router.post("/ask", response_model=ApiResponse)
async def ask(
    req: Request,
    body: CompanionAskRequest,
    user: dict = Depends(get_current_user),
):
    service = CompanionService(db=req.app.state.db)
    answer = await service.answer(context=body.context, question=body.question)
    return {
        "code": 200,
        "data": {"answer": answer},
        "message": "success",
        "meta": {},
    }
